src/main/java/com/minecolonies/client/gui/WindowHutFarmer.java

```diff
@@ -1,198 +1,185 @@
-     *                  This method does not
+     * String which displays the release of a field.
      */
-    private void wheatClicked(Button ignored)
-    {
-        if (building.wheat >= MAX_AMOUNT)
-        {
-            return;
-        }
-        building.wheat++;
-        removeOthers(WHEAT);
-    }
+    private static final String RED_X = "§n§4X";
 
     /**
-     * Action performed when potato button is clicked
-     *
-     * @param ignored   Parameter is ignored, since some actions require a button.
-     *                  This method does not
+     * String which displays adding a field.
      */
-    private void potatoClicked(Button ignored)
-    {
-        if (building.potato >= MAX_AMOUNT)
-        {
-            return;
-        }
-        building.potato++;
-        removeOthers(POTATO);
-    }
+    private static final String APPROVE = "✓";
+
+    /**
+     * Id of the icon inside the GUI.
+     */
+    private static final String TAG_ICON         = "icon";
+
+    /**
+     * Button leading to the previous page.
+     */
+    private Button buttonPrevPage;
 
     /**
-     * Action performed when melon button is clicked
+     * Button leading to the next page.
+     */
+    private Button buttonNextPage;
+
+    /**
+     * List of fields the building seeds.
+     */
+    private List<FieldView> fields = new ArrayList<>();
+
+    /**
+     * ScrollList with the fields.
+     */
+    private ScrollingList   fieldList;
+
+    /**
+     * Constructor for the window of the farmer
      *
-     * @param ignored   Parameter is ignored, since some actions require a button.
-     *                  This method does not
+     * @param building {@link com.minecolonies.colony.buildings.BuildingFarmer.View}
      */
-    private void melonClicked(Button ignored)
+    public WindowHutFarmer(BuildingFarmer.View building)
     {
-        if (building.melon >= MAX_AMOUNT)
-        {
-            return;
-        }
-        building.melon++;
-        removeOthers(MELON);
+        super(building, Constants.MOD_ID + HUT_FARMER_RESOURCE_SUFFIX);
+        registerButton(BUTTON_PREV_PAGE, this::prevClicked);
+        registerButton(BUTTON_NEXT_PAGE, this::nextClicked);
+        registerButton(TAG_BUTTON_ASSIGNMENT_MODE, this::assignmentModeClicked);
+        registerButton(TAG_BUTTON_ASSIGN, this::assignClicked);
     }
 
-
     /**
-     * Action performed when pumpkin button is clicked
-     *
-     * @param ignored   Parameter is ignored, since some actions require a button.
-     *                  This method does not
+     * Fired when assign has been clicked in the field list.
+     * @param button clicked button.
      */
-    private void pumpkinClicked(Button ignored)
+    private void assignClicked(Button button)
     {
-        if (building.pumpkin >= MAX_AMOUNT)
+        int row = fieldList.getListElementIndexByPane(button);
+        FieldView field = fields.get(row);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'field' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)